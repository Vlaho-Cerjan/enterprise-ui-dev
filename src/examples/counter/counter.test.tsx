import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Counter from '.';

test('it should render the component', () => {
  render(<Counter />);
  const currentCount = screen.getByTestId('current-count');
  expect(currentCount).toHaveTextContent('0');
});

test(
  'it should increment when the "Increment" button is pressed',
  async () => {
    const user = userEvent.setup();
    render(<Counter />);

    const currentCount = screen.getByTestId('current-count');
    const incrementButton = screen.getByRole('button', { name: 'Increment' });

    expect(currentCount).toHaveTextContent('0');

    await user.click(incrementButton);
    
    expect(currentCount).toHaveTextContent('1');
  },
);
