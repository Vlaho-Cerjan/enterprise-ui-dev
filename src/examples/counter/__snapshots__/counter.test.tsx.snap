// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it should render the component 1`] = `
<section
  class="flex flex-col gap-8 rounded-lg border-4 border-primary-50 p-8 text-center"
>
  <header>
    <h1>
      Days Since Our Last Gitastropheâ„¢
    </h1>
  </header>
  <div
    class="text-center text-8xl"
    data-testid="current-count"
  >
    0
  </div>
  <div
    class="flex justify-center gap-4"
  >
    <button
      class="btn-danger"
      name="reset"
    >
      Reset
    </button>
    <button
      class="btn-primary"
      name="increment"
    >
      Increment
    </button>
  </div>
</section>
`;
